public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx

```diff
@@ -153,7 +153,7 @@ export const AlertRuleForm: FC<Props> = ({ existing, prefill }) => {
 
   // only reset once we get some value in defaultDsAndQueries.queries, adding this value.
   useEffect(() => {
-    const shouldReset = !existing && !prefill && defaultDsAndQueries.queries;
+    const shouldReset = !existing && !prefill && defaultDsAndQueries.queries && returnTo === '/alerting/list';
```

Comment:
So if `returnTo`=== '/alerting/list` does it mean we're coming from a panel? maybe abstract that into a function for clarity?

Reply:
I updated the pr with another solution